spec/erb_lint/runner_config_spec.rb

```diff
@@ -4,14 +4,85 @@
 
 describe ERBLint::RunnerConfig do
   describe '.default' do
-    subject(:runner_config) { described_class.default }
-
     it 'returns expected class' do
-      expect(subject.class).to(be(described_class))
+      expect(described_class.default.class).to(be(described_class))
+    end
+
+    it 'has default linters enabled' do
+      expect(described_class.default.for_linter('FinalNewline').enabled?).to(be(true))
+    end
+
+    it 'disables default linters if asked to do so' do
+      expect(
+        described_class
+          .default(default_enabled: false)
+          .for_linter('FinalNewline').enabled?
+      ).to(be(false))
+    end
+  end
+
+  describe '.default_for' do
+    context 'without EnableDefaultLinters option' do
+      let(:config_hash) {
+        { foo: true }.deep_stringify_keys
+      }
+
+      let(:expected_result) {
+        {
+          linters: {
+            AllowedScriptType: { enabled: true },
+            ClosingErbTagIndent: { enabled: true },
+            ExtraNewline: { enabled: true },
+            FinalNewline: { enabled: true },
+            NoJavascriptTagHelper: { enabled: true },
+            ParserErrors: { enabled: true },
+            RightTrim: { enabled: true },
+            SelfClosingTag: { enabled: true },
+            SpaceAroundErbTag: { enabled: true },
+            SpaceIndentation: { enabled: true },
+            SpaceInHtmlTag: { enabled: true },
+            TrailingWhitespace: { enabled: true },
+          },
+          foo: true
+        }
+      }
+
+      subject { described_class.default_for(config_hash) }
+      it 'enables all the default linters' do
+        expect(subject.to_hash.to_json).to(include(expected_result.to_json))
+      end
     end
 
-    it 'has FinalNewline enabled' do
-      expect(subject.for_linter('FinalNewline').enabled?).to(be(true))
+    context 'with EnableDefaultLinters set to false' do
+      let(:config_hash) {
+        { linters: { EnableDefaultLinters: false } }.deep_stringify_keys
+      }
+
+      let(:expected_result) {
+        {
+          linters: {
+            AllowedScriptType: { enabled: false },
+            ClosingErbTagIndent: { enabled: false },
+            ExtraNewline: { enabled: false },
+            FinalNewline: { enabled: false },
+            NoJavascriptTagHelper: { enabled: false },
+            ParserErrors: { enabled: false },
+            RightTrim: { enabled: false },
+            SelfClosingTag: { enabled: false },
+            SpaceAroundErbTag: { enabled: false },
+            SpaceIndentation: { enabled: false },
+            SpaceInHtmlTag: { enabled: false },
+            TrailingWhitespace: { enabled: false },
+            EnableDefaultLinters: false
+          }
+        }
+      }
+
+      subject { described_class.default_for(config_hash) }
+      it 'disables all the default linters' do
+        expect(subject.to_hash.to_json).to(include(expected_result.to_json))
```

Comment:
I was gonna do `let(:expected_result) { described_class.default(default_enabled: true) }`. However, `to(include(x)` fails if we exclude `EnableDefaultLinters: false` since it takes into account the entire hash under `linters`. So to avoid specifying each key inside `linters` I just chose to define them all.

Reply:
Since the focus is on linters, perhaps focus on `subject.to_hash["linters"]` (maybe even expose a method for that). I would also go for equality, because this test could allow `MyNewDefaultLinter: { enabled: true }` to creep in and still have the test pass 🙂